src/hooks/usePokemonQuery.jsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { useQuery } from './useQuery';
+import { MAX_CURRENT_POKEMON } from '../components/Pokedex/pokedexConstants';
+import { useHistory, useLocation } from '@docusaurus/router';
+
+const POKEMON_QUERY_PARAMETER_NAME = 'pokemon';
+
+const isValidPokemonDexId = (pokemonDexId) => pokemonDexId >= 0 && pokemonDexId <= MAX_CURRENT_POKEMON;
+
+export const usePokemonQuery = () => {
+  const query = useQuery();
+  const history = useHistory();
+  const location = useLocation();
+
+  const parsedPokemonDexId = Math.floor(Number(query.get(POKEMON_QUERY_PARAMETER_NAME))) || 0;
+  const pokemonDexId = isValidPokemonDexId(parsedPokemonDexId) ? parsedPokemonDexId : 0;
+
+  const setPokemonDexId = (pokemonId) => {
+    const params = new URLSearchParams({ [POKEMON_QUERY_PARAMETER_NAME]: pokemonId });
```

Comment:
Searching only by Pokemon ID isn't something we can do, unless we drop form support in links.

Reply:
every pokemon form has a unique pokemon id, right? - maybe we should also do it like at smogon & have a dedicated name slug for each form: `venusaurmega`, `venusaur` `venusaurdynamax`, ...